### ISSUE TITLE
Suggested a config is not working

### ISSUE THREAD

**Issue by Mattthev** [fireice-uk/xmr-stak-nvidia#31]
The problem from issue #28 is resolved for me, but the suggested config is not working for me. With my previous config file it's working great. I'm getting 237H/s instead of 211H/s in previous update. Is it possible to be 10% more powerful or is my setting just bad and the results are fake?

Suggested config:
`"gpu_threads_conf" : [
  { "index" : 0,    "threads" : 32, "blocks" : 15,    "bfactor" : 6, "bsleep" :  25,    "affine_to_cpu" : false,  },
],`

My config:
`"gpu_threads_conf" : [ 
	{ "index" : 0, "threads" : 20, "blocks" : 15, "bfactor" : 6, "bsleep" :  25, "affine_to_cpu" : false},
],`
I have GTX 860M 2GB Maxwell
Console always close pretty fast so I could make screenshot only. It's not in log file.
![error](https://cloud.githubusercontent.com/assets/6692837/25340065/dc9253d6-2904-11e7-89c5-cacf760a7ee0.jpg)

Thank you guys for resolving the previous problem.

**Comment by fireice-uk** [comment]
@psychocrypt did a lot of work to boost the older cards, so yes, the results are real :).

Autoconfig is a brand new feature so we still need to polish it a bit.

**Comment by psychocrypt** [comment]
@Mattthev Could you please try if `threads : 24`or `threads : 28` is working and which hash rate you get.

**Comment by Mattthev** [comment]
@psychocrypt 28 threads is not working
24 threads 107H/s
21 threads 189H/s
20 threads 237H/s
19 threads 264H/s
18 threads 261H/s
Wow, I was using 20 threads so know I will use 19 threads :D Is it real? It feels like I can't have these hashes :D 
These hashes I have with overclocking to 1232MHz on core and 5522MHz on memory.

**Comment by psychocrypt** [comment]
Yes this is real. I have never seen wrong performance with xmr-stak. If you will see >2kh/s than you should ask:)

**Comment by Mattthev** [comment]
Oh ok :D Thx guys for this super miner. With others I only get about 100 - 150H/s. BTW I'm building my first mining rig with AMD GPUs so I'm excited to use AMD version.

**Comment by fireice-uk** [comment]
@psychocrypt Can we have a look at what happened with the default config here? 32x15 stetting should use 960MB of RAM, no?

**Comment by psychocrypt** [comment]
@fireice-uk Yes the memory used with meta data overhead should be around 1GiB. There is a little bit more memory overhead to run some kernel with lmem. The amount for lmem can only checkt with an additional nvidia library.

First step I will do is to refactor the error handling and check also pointer after cudaMalloc (should not be NULL). Than I will add the memory debug lvl for cases like this.

**Comment by Mattthev** [comment]
So with default clocks and 19 threads I'm getting 206H/s and 600MB of VRAM is used.
[http://gpuz.techpowerup.com/17/04/25/pfk.png](http://gpuz.techpowerup.com/17/04/25/pfk.png)
32 and 28 is not working
24 threads gets 95H/s and 753MB
[http://gpuz.techpowerup.com/17/04/25/4w7.png](http://gpuz.techpowerup.com/17/04/25/4w7.png)
Also the `"bfactor" : 6, "bsleep" :  25,` if both is set to 0 with stock clocks it gets 202H/s.

Hope this can help you guys.
